### PR TITLE
Prevent setting a negative track length

### LINKED ISSUE #1465: Tracks having negative length
If tracks can't be loaded Bass returns a negative value which is not allowed with the most recent changes.

Doc provided by smoogi:
>Bass.ChannelBytes2Seconds
If successful, then the translated length in seconds is returned, else a negative value is returned. Use BASS_ErrorGetCode() to get the error code.

In case of an error (in Bass) these tracks should have a length of 0 instead.

### PR DESCRIPTION
**Pull request by Aergwyn**
Oops, closes #1465

### CODE REVIEW

File: osu.Framework/Audio/Track/TrackBass.cs [hunk 1]
```diff
@@ -93,11 +93,13 @@ void IBassAudio.UpdateDevice(int deviceIndex)
             Trace.Assert(Bass.LastError == Errors.OK);
         }
 
+        private double calculateStreamLength() => Math.Max(Bass.ChannelBytes2Seconds(activeStream, Bass.ChannelGetLength(activeStream)) * 1000, 0);
+
         protected override void UpdateState()
         {
             isRunning = Bass.ChannelIsActive(activeStream) == PlaybackState.Playing;
 
-            double currentTimeLocal = Bass.ChannelBytes2Seconds(activeStream, Bass.ChannelGetPosition(activeStream)) * 1000;
```
Comment (smoogipoo) [inline osu.Framework/Audio/Track/TrackBass.cs]: This used to be `ChannelGetPosition`, but has now been changed to `ChannelGetLength`. This is breaking.

Comment (Aergwyn) [reply osu.Framework/Audio/Track/TrackBass.cs]: Oh right. I'll revert changes on this line entirely as the position actually could be negative I'd assume.

**Comment by smoogipoo** [comment]
Is there a case where this happens and the rest of the code works fine? I feel like the track should be completely unusable at the point where ChannelGetLength returns an error?

**Comment by smoogipoo** [comment]
(i.e. isLoaded should probably be false if that ever happens, via an early return)

**Comment by Aergwyn** [comment]
I'm not sure. Not having any audio device would create said error but not having any tracks loaded without an audio device would make it unplayable? Not that it seems to be fun or even working too well currently.

**Comment by smoogipoo** [comment]
Is there any case beyond not having an audio device? Because from the wording of https://github.com/ppy/osu-framework/pull/1165, that looks to crash the game regardless. A separate fix is necessary for that.

If not, then I don't think this PR is needed, and we should be looking into fixing the no-audio-device issue instead.

**Comment by smoogipoo** [comment]
Given http://www.bass.radio42.com/help/html/63fd986c-70cc-62e1-5287-205f3cadd47d.htm, it looks like the only two errors are invalid handle and notavail - notavail being the cause of attempting to ChannelGetLength on a streaming track, which will need separate handling altogether I believe, if we ever intend to do that.

I don't think this safety is necessary.

**Pull request closed without merging**